### PR TITLE
removed invalid builds and improved tests

### PR DESCRIPTION
**Pull request by caarlos0**
refs https://github.com/goreleaser/goreleaser/issues/191

cc/ @bep

### CODE REVIEW

**Comment by bep** [comment]
Looks good to me.

**Comment by bep** [comment]
But note that I assume you will wire in the implementation, as I currently only see the tests.

**Comment by caarlos0** [comment]
@bep it is wired already...

File: pipeline/build/build.go [hunk 1]
```diff
@@ -36,6 +36,7 @@ func (Pipe) Run(ctx *context.Context) error {
 			goos := goos
 			goarch := goarch
 			if !valid(goos, goarch) {
```
Comment (caarlos0) [inline pipeline/build/build.go]: @bep here.

**Comment by bep** [comment]
OK, then I kind of misunderstood. 

I see OpenBSD + ARM is a valid combo, but that fails on my Mac, and that isn't in the goxc list either.

**Comment by bep** [comment]
I get

```
 -> Building dist/hugo_v0.19.94_openbsd_arm/hugo
# github.com/spf13/hugo/vendor/golang.org/x/sys/unix
vendor/golang.org/x/sys/unix/flock.go:18: undefined: Flock_t
vendor/golang.org/x/sys/unix/sockcmsg_unix.go:36: undefined: Cmsghdr
vendor/golang.org/x/sys/unix/sockcmsg_unix.go:42: undefined: Cmsghdr
```

Which I should report to the Go team (if not already reported), but I'm afraid I will have to wait for a fix, and I don't care about ARM on OpenBSD ...

**Comment by caarlos0** [comment]
@bep goreleaser openbsd/arm pass:

```
❯ go run main.go --skip-validate --skip-publish
Setting defaults
Getting and validating git state
 -> Skipped validations because --skip-validate is set
Loading environment variables
 -> Skipped validations because --skip-validate is set
Building binaries
 -> Building dist/goreleaser_Darwin_i386/goreleaser
 -> Building dist/goreleaser_Darwin_x86_64/goreleaser
 -> Building dist/goreleaser_Linux_i386/goreleaser
 -> Building dist/goreleaser_Linux_x86_64/goreleaser
 -> Building dist/goreleaser_Linux_arm/goreleaser
 -> Building dist/goreleaser_Linux_arm64/goreleaser
 -> Building dist/goreleaser_OpenBSD_i386/goreleaser
 -> Building dist/goreleaser_OpenBSD_x86_64/goreleaser
 -> Building dist/goreleaser_OpenBSD_arm/goreleaser
 -> Building dist/goreleaser_FreeBSD_i386/goreleaser
 -> Building dist/goreleaser_FreeBSD_x86_64/goreleaser
 -> Building dist/goreleaser_FreeBSD_arm/goreleaser
 -> Building dist/goreleaser_Windows_i386/goreleaser.exe
 -> Building dist/goreleaser_Windows_x86_64/goreleaser.exe
Creating archives
 -> Creating dist/goreleaser_FreeBSD_arm.tar.gz
 -> Creating dist/goreleaser_OpenBSD_arm.tar.gz
 -> Creating dist/goreleaser_OpenBSD_x86_64.tar.gz
 -> Creating dist/goreleaser_Darwin_x86_64.tar.gz
 -> Creating dist/goreleaser_Linux_arm64.tar.gz
 -> Creating dist/goreleaser_Darwin_i386.tar.gz
 -> Creating dist/goreleaser_Linux_i386.tar.gz
 -> Creating dist/goreleaser_Windows_i386.tar.gz
 -> Creating dist/goreleaser_Linux_x86_64.tar.gz
 -> Creating dist/goreleaser_Windows_x86_64.tar.gz
 -> Creating dist/goreleaser_Linux_arm.tar.gz
 -> Creating dist/goreleaser_FreeBSD_x86_64.tar.gz
 -> Creating dist/goreleaser_FreeBSD_i386.tar.gz
 -> Creating dist/goreleaser_OpenBSD_i386.tar.gz
 -> Registered artifact goreleaser_OpenBSD_arm.tar.gz
 -> Registered artifact goreleaser_FreeBSD_arm.tar.gz
 -> Registered artifact goreleaser_Windows_i386.tar.gz
 -> Registered artifact goreleaser_Linux_i386.tar.gz
 -> Registered artifact goreleaser_Linux_arm.tar.gz
 -> Registered artifact goreleaser_Darwin_i386.tar.gz
 -> Registered artifact goreleaser_Linux_arm64.tar.gz
 -> Registered artifact goreleaser_FreeBSD_i386.tar.gz
 -> Registered artifact goreleaser_OpenBSD_i386.tar.gz
 -> Registered artifact goreleaser_FreeBSD_x86_64.tar.gz
 -> Registered artifact goreleaser_Darwin_x86_64.tar.gz
 -> Registered artifact goreleaser_Windows_x86_64.tar.gz
 -> Registered artifact goreleaser_Linux_x86_64.tar.gz
 -> Registered artifact goreleaser_OpenBSD_x86_64.tar.gz
Creating Linux packages with fpm
 -> Creating dist/goreleaser_Linux_i386.deb
 -> Creating dist/goreleaser_Linux_arm64.deb
 -> Creating dist/goreleaser_Linux_arm.deb
 -> Creating dist/goreleaser_Linux_x86_64.deb
 -> Registered artifact goreleaser_Linux_x86_64.deb
 -> Registered artifact goreleaser_Linux_i386.deb
 -> Registered artifact goreleaser_Linux_arm.deb
 -> Registered artifact goreleaser_Linux_arm64.deb
Calculating checksums
 -> Checksumming goreleaser_Linux_i386.tar.gz
 -> Checksumming goreleaser_Linux_arm64.deb
 -> Checksumming goreleaser_FreeBSD_arm.tar.gz
 -> Checksumming goreleaser_OpenBSD_arm.tar.gz
 -> Checksumming goreleaser_Windows_i386.tar.gz
 -> Checksumming goreleaser_Linux_arm.tar.gz
 -> Checksumming goreleaser_Darwin_i386.tar.gz
 -> Checksumming goreleaser_Linux_arm64.tar.gz
 -> Checksumming goreleaser_Windows_x86_64.tar.gz
 -> Checksumming goreleaser_Linux_x86_64.tar.gz
 -> Checksumming goreleaser_FreeBSD_i386.tar.gz
 -> Checksumming goreleaser_Linux_x86_64.deb
 -> Checksumming goreleaser_Linux_i386.deb
 -> Checksumming goreleaser_OpenBSD_x86_64.tar.gz
 -> Checksumming goreleaser_OpenBSD_i386.tar.gz
 -> Checksumming goreleaser_Linux_arm.deb
 -> Checksumming goreleaser_FreeBSD_x86_64.tar.gz
 -> Checksumming goreleaser_Darwin_x86_64.tar.gz
 -> Registered artifact goreleaser_checksums.txt
Releasing to GitHub
 -> Skipped because --skip-publish is set
Creating homebrew formula
 -> Skipped because --skip-publish is set
Done!
```

**Comment by bep** [comment]
Sure, but you don't use the ` Flock_t` or `Cmsghdr` syscall...

**Comment by caarlos0** [comment]
@bep yeah, just saw your comment.

BTW, there is a similar issue: https://github.com/golang/go/issues/10087

**Comment by bep** [comment]
>there is a similar issue

I would not call it very similar; there is a difference: I don't care about ARM on OpenBSD.

**Comment by caarlos0** [comment]
@bep just removed more combinations, can you take a look?

**Comment by bep** [comment]
I will test it and report back.

**Comment by bep** [comment]
Yes! Now this builds to success with Go 1.8.1 with Hugo:

```
  goos:
    - darwin
    - linux
    - windows
    - freebsd
    - netbsd
    - openbsd
    - dragonfly
  goarch:
    - amd64
    - 386
    - arm
    - arm64
```

Which gives me this list:

```
3964667 Apr 21 19:13 hugo_v0.18.95_Linux_arm.tar.gz
3944955 Apr 21 19:13 hugo_v0.18.95_FreeBSD_32-bit.tar.gz
4422028 Apr 21 19:13 hugo_v0.18.95_macOS_64-bit.tar.gz
4140288 Apr 21 19:13 hugo_v0.18.95_macOS_32-bit.tar.gz
4218866 Apr 21 19:13 hugo_v0.18.95_Windows_64-bit.tar.gz
3941258 Apr 21 19:13 hugo_v0.18.95_Windows_32-bit.tar.gz
4204813 Apr 21 19:13 hugo_v0.18.95_OpenBSD_64-bit.tar.gz
3939779 Apr 21 19:13 hugo_v0.18.95_OpenBSD_32-bit.tar.gz
3966504 Apr 21 19:13 hugo_v0.18.95_NetBSD_arm.tar.gz
4206335 Apr 21 19:13 hugo_v0.18.95_NetBSD_64-bit.tar.gz
3940261 Apr 21 19:13 hugo_v0.18.95_NetBSD_32-bit.tar.gz
3927756 Apr 21 19:13 hugo_v0.18.95_Linux_arm64.tar.gz
4212286 Apr 21 19:13 hugo_v0.18.95_Linux_64-bit.tar.gz
3947667 Apr 21 19:13 hugo_v0.18.95_Linux_32-bit.tar.gz
3969239 Apr 21 19:13 hugo_v0.18.95_FreeBSD_arm.tar.gz
4212642 Apr 21 19:13 hugo_v0.18.95_FreeBSD_64-bit.tar.gz
4207382 Apr 21 19:13 hugo_v0.18.95_DragonFlyBSD_64-bit.tar.gz
3828804 Apr 21 19:13 hugo_v0.18.95_Linux_arm64.deb
3832232 Apr 21 19:13 hugo_v0.18.95_Linux_arm.deb
3830168 Apr 21 19:13 hugo_v0.18.95_Linux_32-bit.deb
4145218 Apr 21 19:13 hugo_v0.18.95_Linux_64-bit.deb
   2077 Apr 21 19:13 hugo_checksums.txt
```

Which, drum roll, looks just about perfect (I will have to adjust the replacements a little to get it in line with the current scheme, but that is my plate).

Thanks alot for the quick support, really appreciate it!

**Comment by caarlos0** [comment]
Nice @bep :D

Thanks for reporting all those problems and making it easier for future users! 🍻

**Pull request merged**